samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/SectionContentsBean.java

```diff
@@ -391,101 +419,19 @@ public SectionContentsBean(SectionDataIfc section)
 
   public void setMetaData(SectionDataIfc section)
   {
-    if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE) != null)
-    {
+    if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE) != null) {
       Integer authortype = new Integer(section.getSectionMetaDataByLabel(
         SectionDataIfc.AUTHOR_TYPE));
       setSectionAuthorType(authortype);
 
       if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE).equals(
-        SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOL.toString()))
-      {
-        if (section.getSectionMetaDataByLabel(SectionDataIfc.
-                                              NUM_QUESTIONS_DRAWN) != null)
-        {
-          Integer numberdrawn = new Integer(section.getSectionMetaDataByLabel(
-            SectionDataIfc.NUM_QUESTIONS_DRAWN));
-          setNumberToBeDrawn(numberdrawn);
-        }
-
-        if (section.getSectionMetaDataByLabel(SectionDataIfc.
-                                              POOLID_FOR_RANDOM_DRAW) != null)
-        {
-          Long poolid = new Long(section.getSectionMetaDataByLabel(
-            SectionDataIfc.POOLID_FOR_RANDOM_DRAW));
-          setPoolIdToBeDrawn(poolid);
+        SectionDataIfc.RANDOM_DRAW_FROM_QUESTIONPOOL.toString())) {
+          setMetadataRandowDraw(section);
+      } else if (section.getSectionMetaDataByLabel(SectionDataIfc.AUTHOR_TYPE).equals(
+          SectionDataIfc.FIXED_AND_RANDOM_DRAW_FROM_QUESTIONPOOL.toString())) {
+          setMetadataFixed(section);
+          setMetadataRandowDraw(section);
         }
```

Comment:
Fix spacing (I know it's pre-existing but looks weird)

Reply:
Put all the if on one line